includes/gateway/class-omise-payment-truemoney.php

```diff
@@ -82,22 +91,51 @@ public function charge($order_id, $order)
 
 	public function get_charge_request($order_id, $order)
 	{
-		$phoneOption = $_POST['omise_phone_number_default'];
-		$isPhoneOptionChecked = isset($phoneOption) && 1 == $phoneOption;
-		$phone_number = $isPhoneOptionChecked ?
-			$order->get_billing_phone() :
-			sanitize_text_field( $_POST['omise_phone_number'] );
-
 		$requestData = $this->build_charge_request(
 			$order_id,
 			$order,
 			$this->source_type,
 			$this->id . '_callback'
 		);
-		$requestData['source'] = array_merge($requestData['source'], [
-			'phone_number' => $phone_number
-		]);
+
+		if (self::WALLET === $this->source_type) {
+			$phoneOption = $_POST['omise_phone_number_default'];
+			$isPhoneOptionChecked = isset($phoneOption) && 1 == $phoneOption;
+			$phone_number = $isPhoneOptionChecked ?
```

Comment:
similar to my comment re: function naming

Reply:
Thanks. Will change it to snake_case.